flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/cleanup/DispatcherResourceCleanerFactoryTest.java

```diff
@@ -192,7 +192,6 @@ private void assertLocalCleanupNotTriggered() {
         assertThat(jobManagerRunnerRegistryLocalCleanupFuture).isNotDone();
         assertThat(jobGraphWriterLocalCleanupFuture).isNotDone();
         assertThat(blobServer.getLocalCleanupFuture()).isNotDone();
-        assertThat(jobManagerMetricGroup.numRegisteredJobMetricGroups()).isEqualTo(1);
```

Comment:
is it not triggered because the global cleanup runs first?

Reply:
It just seems a bit weird because conceptually any scenario in which a global resource is cleaned up should also try cleaning up the local resource?

Reply:
We had this discussion previously. We decided to not model a dependency between the local and the global cleanup into the cleanable resources. That's knowledge that's held by the Dispatcher how he (?) manages cleanups.

Reply:
There are also cases where the local and global cleanup shall be performed in a synchronized fashion (e.g. `BlobServer`) which required to have the two interfaces (i.e. the two different types of cleanup methods) being operatable independent from each other. I updated the JavaDoc of `GloballyCleanableResource` and `LocallyCleanableResource` to reflect. The decision on what method to call is now based on whether the job reaches a locally terminal or globally terminal state. I think that makes the intention of these two interfaces clearer. WDYT?